recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -71,6 +71,16 @@ def _min_cppstd(self):
             return 14
         return 11
 
+    @property
+    def _default_cppstd(self):
+        return default_cppstd(self).replace("gnu", "")
```

Comment:
TODO: check why apple-clang fails to build with `default_cppstd` or add this:
```suggestion
        if self.options.with_stl == "default":
            return default_cppstd(self).replace("gnu", "")
        return self._min_cppstd
```

Reply:
@jcar87 `default_cppstd` for apple-clang13 profile returns `gnu98` - is this correct?

<details><summary>apple-clang13</summary>
<p>

```
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
[options]
opentelemetry-cpp:shared=False
```

</p>
</details>

Reply:
Hi @steinerthomas, yes, this is correct. Unfortuntately apple-clang has not changed the default cppstd between versions, so the default cppstd for every version is still gnu98

Reply:
Hi @RubenRBS, thanks for your answer. Is there a utility where I can compare cppstd versions? This would be helpful here: if _min_cppstd greater than default_cppstd then return _min_cppstd  else return default_cppstd